### DISCUSSION TITLE
Github pages is deploying READme file

### DISCUSSION THREAD

**Discussion by ChethanHR10** [community/community#128670] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

https://github.com/ChethanHR10/Weather-webpage
I tried to deploy this project using GitHub pages.
I tried everything gh -pages etc., but still not able to deploy this react.js project. Please help me with this problem.

Also, I tried this in vercel and there it is deploying but the whole page is blank(white).

**Comment by davevad93** [comment]
Hi @ChethanHR10 ,

The issue occurs because your `index.html` is inside the `public` folder, for GitHub Pages sites to display correctly the `index.html` must be in the root of the repo. Note that you may need to update the paths of your images or scripts inside the `html`. The structure should be as follows:

/[Weather-webpage](https://github.com/ChethanHR10/Weather-webpage)
│
├── public
├── src
├── .gitignore
├── index.html
├──package-lock.json
├── package.json
└── README.md

**Reply by davevad93** [reply]
@ChethanHR10 , look inside your `public` folder, there's a file named `index.html`, put that file in the root of your repo.

**Reply by ChethanHR10** [reply]
https://chethanhr10.github.io/Weather-webpage/

I have set it to main/root but a new problem is there (it is working fine in my local hosting).

**Reply by ChethanHR10** [reply]
![Screenshot 2024-06-15 141314](https://github.com/community/community/assets/124980611/3d8b7119-df48-449e-8b13-f528785dad7a)
This is from brave browser
![image](https://github.com/community/community/assets/124980611/f266a527-c580-47af-b00b-66694f46544d)

**Reply by davevad93** [reply]
Change the following lines:
```html
<link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
<link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
<link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
```    
To this:
```html
<link rel="icon" href="public/favicon.ico" />
<link rel="apple-touch-icon" href="public/logo192.png" />
<link rel="manifest" href="public/manifest.json" />
```

**Reply by ChethanHR10** [reply]
the errors in the browser is eliminated but the page is still white
![image](https://github.com/community/community/assets/124980611/278b51ee-85a1-40f6-84e6-ed82d3f6d92f)

**Reply by davevad93** [reply]
In `package.json` change the `homepage` value from:
```json
"homepage": "https://github.com/ChethanHR10/Weather-webpage",
``` 
to:
```json
"homepage": "https://chethanhr10.github.io/Weather-webpage",
```  
Let me know if it fixes it!

**Reply by ChethanHR10** [reply]
No sir!!!! still the same.

**Reply by davevad93** [reply]
At this point I think it must be something with the configuration, honestly I've never deployed on `GitHub Pages` a `React` app so I'm not sure what might be the issue here. I'm sorry if i wasn't able to help you properly, take a look at this resources, they might be useful:

- [Getting blank page after react app publish in github](https://stackoverflow.com/questions/54427793/getting-blank-page-after-react-app-publish-in-github)
- [React app deployment on GitHub Pages displays a blank page](https://stackoverflow.com/questions/75860344/react-app-deployment-on-github-pages-displays-a-blank-page)
- [Blank page when deploying React App to gh pages](https://stackoverflow.com/questions/62242263/blank-page-when-deploying-react-app-to-gh-pages)
- [React app deployment on github pages shows a blank page](https://stackoverflow.com/questions/68025460/react-app-deployment-on-github-pages-shows-a-blank-page)

**Reply by ChethanHR10** [reply]
I appreciate your efforts, Thank you so much sir!!!

**Comment by LiteBrite82** [comment]
Looking at your file structure in the main branch, I dont think you have it set up correctly. Are you using "Create-React-App"?

Routing is typically done in app.js; not index.js. components and pages should be in their own files/folders. 

index.html should be in the public folder.

**Reply by davevad93** [reply]
Hey @LiteBrite82 , I didn't know `index.html` for a `React` project has to be in the `public` folder, I thought that by default it always has to be in the root of the repo, I learned something new, thanks a lot!

**Reply by davevad93** [reply]
I got it all wrong by the way, sorry again! @ChethanHR10  😅

**Reply by LiteBrite82** [reply]
@davevad93 

yeah, for source code its gotta be in public. But for the built app, yes it needs to be in root.  
Your walkthrough detail is on point though! 

@ChethanHR10 

The source code will run fine locally, but the builds are failing because of issues in the code. 

You need to fix the code in index.js and in app.js. I don't believe you can have 2 returns in the same component. Also you would need to import the navbar component from somewhere, along with BrowserRouter. 

Honestly, before going too much further, I would consider using Vite or or a framework like it instead of Create-React-App. CRA has ben deprecated and wont hold up for much longer. 

@noorjsdivs made a great [YouTube video](https://www.youtube.com/watch?v=peS-Y3BcfUo) on how to set up Vite, react-router-dom, and tailwind. (Thanks Noor!!!).

Keep up the hard work, ChethanHR10; you are getting close!!!

**Reply by ChethanHR10** [reply]
@LiteBrite82  
Thank you so much for your help, i really appreciate it 😄.
@davevad93 
It's ok, we learn new things everyday.
Thank you again.
